great_expectations/core/usage_statistics/anonymizers/profiler_run_anonymizer.py

```diff
@@ -0,0 +1,251 @@
+    ColumnDomainBuilder,
+)
+from great_expectations.rule_based_profiler.domain_builder.domain_builder import (
+    DomainBuilder,
+)
+from great_expectations.rule_based_profiler.domain_builder.simple_column_suffix_domain_builder import (
+    SimpleColumnSuffixDomainBuilder,
+)
+from great_expectations.rule_based_profiler.domain_builder.simple_semantic_type_domain_builder import (
+    SimpleSemanticTypeColumnDomainBuilder,
+)
+from great_expectations.rule_based_profiler.domain_builder.table_domain_builder import (
+    TableDomainBuilder,
+)
+from great_expectations.rule_based_profiler.expectation_configuration_builder.default_expectation_configuration_builder import (
+    DefaultExpectationConfigurationBuilder,
+)
+from great_expectations.rule_based_profiler.expectation_configuration_builder.expectation_configuration_builder import (
+    ExpectationConfigurationBuilder,
+)
+from great_expectations.rule_based_profiler.parameter_builder.metric_multi_batch_parameter_builder import (
+    MetricMultiBatchParameterBuilder,
+)
+from great_expectations.rule_based_profiler.parameter_builder.numeric_metric_range_multi_batch_parameter_builder import (
+    NumericMetricRangeMultiBatchParameterBuilder,
+)
+from great_expectations.rule_based_profiler.parameter_builder.parameter_builder import (
+    ParameterBuilder,
+)
+from great_expectations.rule_based_profiler.parameter_builder.regex_pattern_string_parameter_builder import (
+    RegexPatternStringParameterBuilder,
+)
+from great_expectations.rule_based_profiler.parameter_builder.simple_date_format_string_parameter_builder import (
+    SimpleDateFormatStringParameterBuilder,
+)
+from great_expectations.util import deep_filter_properties_iterable
+
+logger = logging.getLogger(__name__)
+
+
+class ProfilerRunAnonymizer(Anonymizer):
+    def __init__(self, salt: Optional[str] = None) -> None:
+        super().__init__(salt=salt)
+
+        # ordered bottom up in terms of inheritance order
+        self._ge_domain_builders = [
+            SimpleColumnSuffixDomainBuilder,
+            SimpleSemanticTypeColumnDomainBuilder,
+            ColumnDomainBuilder,
+            TableDomainBuilder,
+            DomainBuilder,
+        ]
+        self._ge_parameter_builders = [
+            MetricMultiBatchParameterBuilder,
+            NumericMetricRangeMultiBatchParameterBuilder,
+            RegexPatternStringParameterBuilder,
+            SimpleDateFormatStringParameterBuilder,
+            ParameterBuilder,
+        ]
+        self._ge_expectation_configuration_builders = [
+            DefaultExpectationConfigurationBuilder,
+            ExpectationConfigurationBuilder,
+        ]
+        self._ge_expectation_types = set(list_registered_expectation_implementations())
+
+        self._salt = salt
+        self._batch_request_anonymizer = BatchRequestAnonymizer(self._salt)
+
+    def anonymize_profiler_run(self, *args: List[Any], **kwargs: dict) -> dict:
```

Comment:
Would it make more sense here to take the ProfilerConfig or BaseRuleBasedProfiler object instead of a dict? That way we can solidify the interface and not have to pull data via named kwargs.